rest-service/manager_rest/rest/resources_v2/events.py

```diff
@@ -112,11 +108,24 @@ def get(self, _include=None, filters=None,
             for event in select_query.params(**params).all()
         ]
 
+        events_total = None
+        logs_total = None
+        for r in results:
```

Comment:
I'm not sure I understand what's going on here - why are we iterating over all of those results? Is the `full_count` label added to _each_ result? If so, that's kind of weird, and also might make us iterate over all of this loop for nothing (for example, if all we have are 10000 events, with no logs)

Reply:
We're iterating over all of the results because unfortunately we currently obtain the events and the logs in two separate queries.

I can think of a couple of ways to make this more efficient:
1. Probably the fastest and quite possibly the best approach, but also more complex and potentially error prone (though probably lower technical debt longer term): Retrieve both events and logs in a single query instead of two sub-queries, meaning that only the first result's full_count field needs to be retrieved.

2. Pick up from the parameters whether we need both logs_total and events_total, and set either of them to 0 if they're not needed before starting iterating over the results. This would mean that if you only asked for events then we'd only get the total for events, so we should stop after checking one record. Otherwise, we'd have to scan all of them anyway.

Reply:
The second optimization is fine enough, but of course, the worst case scenario is left unaddressed. 
A general question - is there no way for us to use our default means of getting this data (i.e. the storage manager)? Why are we doing this whole crazy thing? Performance?

Reply:
The worst case scenario is /marginally/ improved. In the (unlikely) case that we have 10,000 events and 1 log (or 10,000 logs and 1 event), we'll indeed have to iterate over all of them to get the details.

This is for correctness more than performance- actually getting the total with the specified filters rather than using two different queries performed at slightly different times to get an incorrect total count.